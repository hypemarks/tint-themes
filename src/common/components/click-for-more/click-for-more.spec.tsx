import { render, fireEvent } from '@testing-library/react';
import React from 'react';
import ClickForMore from './click-for-more';

const onNewContent = jest.fn();
const fetchNextPage = jest.fn();

describe('Test <ClickForMore /> Component', () => {
  it('should render snapshot', () => {
    const wrapper = render(<ClickForMore onNewContent={onNewContent} fetchNextPage={fetchNextPage} />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
  it('should render ClickForMore component', () => {
    const wrapper = render(<ClickForMore onNewContent={onNewContent} fetchNextPage={fetchNextPage} />);
    const clickForMore = wrapper.queryByTestId('click-for-more');
    expect(clickForMore).toBeInTheDocument();
    expect(clickForMore).toHaveTextContent('click for more');
  });

  it('should fire handleOnClick', () => {
    const wrapper = render(<ClickForMore onNewContent={onNewContent} fetchNextPage={fetchNextPage} />);
    const clickForMore = wrapper.getByText('click for more');
    fireEvent.click(clickForMore);
    expect(onNewContent).toBeCalled();
  });
});
