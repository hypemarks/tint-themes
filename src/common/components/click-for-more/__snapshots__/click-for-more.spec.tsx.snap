// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <ClickForMore /> Component should render snapshot 1`] = `
<DocumentFragment>
  <button
    class="click-for-more button"
    data-testid="click-for-more"
    tabindex="0"
    type="button"
  >
    click for more
  </button>
</DocumentFragment>
`;
